pugl/detail/x11.c

```diff
@@ -597,6 +597,9 @@ puglPollEvents(PuglWorld* world, const double timeout)
 	sendAllRedisplayEvents(world);
 	XFlush(world->impl->display);
 
+	if (XEventsQueued(world->impl->display, QueuedAlready)) {
```

Comment:
When would this happen?

Reply:
Perhaps XFlush does also read some events from the fd itself into internal buffer? There are also other xlib function calls that are reading from the x11 connection. It's not specified which xlib function does some reading from the fd, so it's always a good idea to check if X11 read something before doinng a `select' on the fd.